internal/cache/file/cache_handle.go

```diff
@@ -148,28 +149,47 @@ func (fch *CacheHandle) Read(ctx context.Context, object *gcs.MinObject, offset
 	}
 
 	// We are here means, we have the data downloaded which kernel has asked for.
-	_, err = fch.fileHandle.Seek(offset, 0)
-	if err != nil {
-		errMsg := fmt.Sprintf("%s: while seeking for %d offset in local file: %v", util.ErrInSeekingFileHandleMsg, offset, err)
-		return 0, errors.New(errMsg)
-	}
-
-	n, err = io.ReadFull(fch.fileHandle, dst)
+	n, err = fch.fileHandle.ReadAt(dst, offset)
+	requestedNumBytes := int(requiredOffset - offset)
 	// Returns io.ErrUnexpectedEOF when buffer size is more than content to be read.
 	if err == io.EOF || err == io.ErrUnexpectedEOF {
 		err = nil
 	}
 	if err != nil {
 		errMsg := fmt.Sprintf("%s: while reading from %d offset of the local file: %v", util.ErrInReadingFileHandleMsg, offset, err)
 		return 0, errors.New(errMsg)
+	} else if n != requestedNumBytes {
```

Comment:
Do we know, when does it happen?

Reply:
No, I added this check because if this happens, we will never know. And if this never happens, then also there is no harm in having simple additional check.

Reply:
one thing that I thought but didn't created the scenario - if a file is truncated externally such that it contains data till offset + x but x is less than len(dst)

Reply:
I think, it's good to document this case around the check.